lib/gingr/geoserver_publisher.rb

```diff
@@ -1,55 +1,110 @@
-        'url' => rest_url(uri),
-        'user' => uri.user,
-        'password' => uri.password.to_s,
-      })
-    end
+    DEFAULT_REMOTE_ROOT = '/srv/geofiles'
+    DEFAULT_WORKSPACE = 'UCB'
+
+    attr_reader :connection
+    attr_reader :remote_root
+    attr_reader :workspace_name
+
+    class << self
+      def publish_inventory(inventory, geoserver_url: nil, geoserver_secure_url: nil)
+        if !inventory[:public].empty?
+          public_publisher = new(geoserver_url)
+          public_publisher.batch_publish(inventory[:public])
+        end
+
+        if !inventory[:ucb].empty?
+          secure_publisher = new(geoserver_secure_url, default: :geoserver_secure_url)
+          secure_publisher.batch_publish(inventory[:ucb])
+        end
+      end
 
-    def update(filename)
-      name = File.basename(filename, '.*')
-      filepath = "file:///srv/geofiles/berkeley-#{name}/#{filename}"
-      File.extname(filename).downcase == '.shp' ? publish_shapefile(filepath, name) : publish_geotiff(filepath, name)
-    rescue Geoserver::Publish::Error => e
-      logger.error("Publish Geoserver error: #{filename} -- #{e.inspect}")
-      raise
+      def parse_connection_string(geoserver_baseurl)
+        uri = URI.parse(geoserver_baseurl)
+        uri.path << '/' unless uri.path.end_with? '/'
+        uri.path << 'rest/' unless uri.path.end_with? 'rest/'
+
+        return URI::Generic.build(
+          scheme: uri.scheme,
+          host: uri.host,
+          port: uri.port == uri.default_port ? nil : uri.port,
+          path: uri.path,
+          fragment: uri.fragment,
+          query: uri.query,
+        ).to_s, uri.user, uri.password
+      end
     end
 
-    def publish_shapefile(filepath, name)
-      Geoserver::Publish.shapefile(connection: @conn, workspace_name: 'UCB', file_path: filepath,
-                                   id: name, title: name)
+    def initialize(conn = nil, default: nil, remote_root: nil, workspace_name: nil)
+      conn ||= Gingr::Config.getopt(default || :geoserver_url)
+
+      # Coerce a connection string into an actual connection object
+      if conn.kind_of? String
+        rest_url, user, password = self.class.parse_connection_string(conn)
+        conn = Geoserver::Publish::Connection.new({
+          'url' => rest_url,
+          'user' => user,
+          'password' => password,
+        })
+      end
+
+      @connection = conn
+      @remote_root = (remote_root || DEFAULT_REMOTE_ROOT).chomp '/'
+      @workspace_name = workspace_name || DEFAULT_WORKSPACE
     end
 
-    def publish_geotiff(filepath, name)
-      Geoserver::Publish.geotiff(connection: @conn, workspace_name: 'UCB', file_path: filepath, id: name,
-                                 title: name)
+    def batch_publish(filenames)
+      filenames.each(&method(:publish))
     end
 
-    def batch_update(filename_list)
-      filename_list.each { |filename| update(filename) }
+    def publish(filename)
```

Comment:
I renamed this from "update" to "publish" to maintain consistency, since the underlying methods are all called "publish_<type>".